src/codegen.cpp

```diff
@@ -9173,6 +9173,9 @@ jl_llvm_functions_t jl_emit_code(
         jl_static_show((JL_STREAM*)STDERR_FILENO, jl_current_exception());
         jl_printf((JL_STREAM*)STDERR_FILENO, "\n");
         jlbacktrace(); // written to STDERR_FILENO
+#ifndef JL_NDEBUG
+        abort();
```

Comment:
Alternative, this could be hidden behind an env var, e.g., `JULIA_ABORT_ON_INTERNAL_ERROR`. We could additionally use that to abort when there's an IR verification error, making PkgEval's life easier without potentially complicating compiler development.